module/data/item/templates/physical-item.mjs

```diff
@@ -212,22 +212,35 @@ export default class PhysicalItemTemplate extends SystemDataModel {
 
   /* -------------------------------------------- */
 
+  /** @inheritDoc */
+  async _preUpdate(changed, options, user) {
+    await super._preUpdate(changed, options, user);
+    if ( foundry.utils.hasProperty(changed, "system.container") ) {
+      options.formerContainer = (await this.parent.container)?.uuid;
+    }
+  }
+
+  /* -------------------------------------------- */
+
   /** @inheritDoc */
   _onCreate(data, options, userId) {
+    super._onCreate(data, options, userId);
     this._renderContainers();
   }
 
   /* -------------------------------------------- */
 
   /** @inheritDoc */
   _onUpdate(changed, options, userId) {
+    super._onUpdate(changed, options, userId);
     this._renderContainers({ formerContainer: options.formerContainer });
   }
 
   /* -------------------------------------------- */
 
   /** @inheritDoc */
   _onDelete(options, userId) {
+    super._onDelete(options, userId);
     this._renderContainers();
   }
```

Comment:
Some of these were `async` for sub-types. Is it different for the template mixins? We don't necessarily have to `await` the `super` call, especially if this is render stuff, but we should make sure the signature at least matches if we are doing `@inheritDoc`.

Reply:
The signature for these methods isn't `async` in `TypeDataModel`, so that's what I've been going off of. I think most of the `async` `_on*` method can sync instead.